packages/cli/cli.js

```diff
@@ -39,14 +40,16 @@ const args = minimist(process.argv.slice(2), {
 })
 
 if (args.version) {
-  const version = JSON.parse(await readFile(join(import.meta.url, 'package.json'))).version
+  const { version } = await asyncPipe(
+    () => join(import.meta.url, 'package.json'),
+    readFile,
+    JSON.parse
+  )()
```

Comment:
```suggestion
  const packageJsonFile = await readFile(join(import.meta.url, 'package.json')
  const packageJson = JSON.parse(packageJsonFile)
  const version = packageJson.version
```

When I see functions like `asyncPipe` in the code I have to go and check what it does. If you want to make it more readable just split it on a few variables with good names. async/await mechanism is a very good feature to write asynchronous code in a readable synchronous way.

Reply:
The pipe is a popular function which is used in functional programming, in Unix, etc., but if you think so - no problem